### ISSUE TITLE
[bug] Warning in workflow about set-output

### ISSUE THREAD

**Issue by mrgharabaghi** [actions/download-artifact#185]
### What happened?

Warning: `The "set-output" command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/`

### What did you expect to happen?

Downloading the artifact without deprecation warning.

### How can we reproduce it?

```yml
on: push
jobs:
  build-node:
    runs-on: ubuntu-22.04
    container: node:18
    steps:
      # -----------------------------------------------
      - run: node --version
      - run: npm --version
      - uses: actions/checkout@v3 # uses
      - run: npm i
      - run: mkdir -p path/to/artifact
      - run: echo hello > path/to/artifact/world.txt
      # -----------------------------------------------
      - name: Copy File
        working-directory: work
        run: |
          npm i
          npm run build
      # -----------------------------------------------
      - uses: actions/upload-artifact@v3 # uses
        with:
          name: my-artifact-1
          path: path/to/artifact/world.txt
          retention-days: 1
      # -----------------------------------------------
      - uses: actions/upload-artifact@v3 # uses
        with:
          name: my-artifact-2
          path: work/built
          retention-days: 1
      # -----------------------------------------------
      - name: Download all workflow run artifacts
        uses: actions/download-artifact@v3
      # -----------------------------------------------
      - name: Display structure of downloaded files
        run: |
          tar -C ./my-artifact-1 -cvf my-artifact-1.tar.gz .
          tar -C ./my-artifact-2 -cvf my-artifact-2.tar.gz .
```

### What version of the action are you using?

v3.0.0

### What are your runner environments?

linux

**Comment by IdanYaffe** [comment]
Please fix this.
It's causing a lot of white noise when looking at the workflows.

**Comment by mquinzin** [comment]
Also interested ina fix

**Comment by scripthunter7** [comment]
This line causes it, I guess:
https://github.com/actions/download-artifact/blob/782e5ae9ea8551dd997e5c4dd22499cb62797251/src/download-artifact.ts#L54

According to the [link in the warning](https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/), at least version `1.10.0` from the `@actions/core` package is required, but the [currently used version](https://www.npmjs.com/package/@actions/core/v/1.2.6) is ~2 years old:
https://github.com/actions/download-artifact/blob/076f0f7dd036d87e8e04f5f00d614e790308961b/package.json#L32

**Comment by jsoref** [comment]
#178 / #183 would resolve this.

**Comment by hakostra** [comment]
This is very annoying, also waiting for a fix...

**Comment by Akitakek** [comment]
Cannot have my package uploaded to deployment without downloading the artifacts first, and without a fix to this, can't do so.

**Comment by konradpabjan** [comment]
This was fixed by in October with the `v3.0.1` release: https://github.com/actions/download-artifact/releases/tag/v3.0.1
